### PR TITLE
“Facts on the ground” XSD matching ios-api-docs

### PR DESCRIPTION
**Pull request by lobrien**
Relates to https://github.com/xamarin/ios-api-docs/issues/44 & https://github.com/xamarin/DocWriter/issues/11

Closes https://github.com/mono/api-doc-tools/issues/71

With this, it would be possible to add, e.g., `xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation=“https://github.com/xamarin/api-doc-tools/mdoc/Resources/monodoc-ecma.xsd”` to the root `Type` element.

### CODE REVIEW

**Comment by lobrien** [comment]
It's worth noting that this is by no means a "perfect" monodoc-xsd. It is definitely bloated and the same exercise of applying it to all the XML in our larger codebases should be done. After we do that, we should start tightening it up.

File: mdoc/Resources/monodoc-ecma.xsd [hunk 1]
```diff
@@ -71,7 +78,6 @@ add masterdoc support?
   <xs:element name="permille" />
   <xs:element name="pi" />
   <xs:element name="pre" />
-  <xs:element name="ReturnType" type="xs:string" />
```
Comment (joelmartinez) [inline mdoc/Resources/monodoc-ecma.xsd]: Just curious ... why is this removing `ReturnType` ?

File: mdoc/Resources/monodoc-ecma.xsd [hunk 2]
```diff
@@ -152,6 +180,7 @@ add masterdoc support?
       <xs:choice minOccurs="0" maxOccurs="unbounded">
         <xs:element ref="block" />
         <xs:element ref="c" />
+        <!-- bryan: should this be allowed? code without example? -->
```
Comment (joelmartinez) [inline mdoc/Resources/monodoc-ecma.xsd]: We can probably remove the "named" comments

File: mdoc/Resources/monodoc-ecma.xsd [hunk 3]
```diff
@@ -162,6 +191,8 @@ add masterdoc support?
         <xs:element ref="sup" />
         <xs:element ref="pi" />
         <xs:element ref="typeparamref" />
+        <!-- bryan -->
```
Comment (joelmartinez) [inline mdoc/Resources/monodoc-ecma.xsd]: another candidate for removal

File: mdoc/Resources/monodoc-ecma.xsd [hunk 4]
```diff
@@ -100,7 +118,8 @@ add masterdoc support?
         <xs:element ref="AssemblyCulture" minOccurs="0" />
         <xs:element ref="Attributes" minOccurs="0" />
       </xs:sequence>
-    </xs:complexType>
+      <xs:attribute ref="apistyle"/>
+     </xs:complexType>
```
Comment (jonpryor) [inline mdoc/Resources/monodoc-ecma.xsd]: This indentation looks wrong.

File: mdoc/Resources/monodoc-ecma.xsd [hunk 5]
```diff
@@ -478,25 +521,15 @@ add masterdoc support?
       </xs:sequence>
       <xs:attribute ref="MemberName" />
       <xs:attribute ref="Deprecated" />
-    </xs:complexType>
-  </xs:element>
-
-  <xs:element name="MemberGroup">
```
Comment (jonpryor) [inline mdoc/Resources/monodoc-ecma.xsd]: I don't think we want to remove `<MemberGroup/>`. See #35, #36.

File: mdoc/Resources/monodoc-ecma.xsd [hunk 6]
```diff
@@ -121,6 +140,15 @@ add masterdoc support?
       </xs:sequence>
     </xs:complexType>
   </xs:element>
+  
+  <!-- License reference -->
+  <xs:element name="attribution">
```
Comment (jonpryor) [inline mdoc/Resources/monodoc-ecma.xsd]: Why move this declaration?

File: mdoc/Resources/monodoc-ecma.xsd [hunk 7]
```diff
@@ -121,6 +140,15 @@ add masterdoc support?
...
+    <xs:complexType>
+    <xs:attribute ref="license"/>
```
Comment (jonpryor) [inline mdoc/Resources/monodoc-ecma.xsd]: This should be indented another "tab stop" (2 spaces).

File: mdoc/Resources/monodoc-ecma.xsd (hunk 5 above)
Comment (joelmartinez) [inline mdoc/Resources/monodoc-ecma.xsd]: Ahh, yes ... will definitely need to retain this

**Comment by directhex** [comment]
build

**Pull request closed without merging**